plugins/PrivacyManager/tests/Integration/PrivacyManagerConfigTest.php

```diff
@@ -126,8 +129,40 @@ public function testSetTrackerCacheContentShouldGetValuesFromConfig()
     {
         Option::set('PrivacyManager.ipAddressMaskLength', '232');
 
-        $content = $this->config->setTrackerCacheGeneral(array('existingEntry' => 'test'));
+        $trackerCache = ['existingEntry' => 'test'];
+        $content = $this->config->setTrackerCache($trackerCache);
 
         $this->assertEquals(232, $content['PrivacyManager.ipAddressMaskLength']);
     }
+
+    public function testSetTrackerCacheContentShouldGetValuesFromConfigForSite()
+    {
+        Option::set('PrivacyManager.idSite(1).ipAddressMaskLength', '345');
+
+        $trackerCache = ['existingEntry' => 'test'];
+        $this->config->setIdSite(1);
+        $content = $this->config->setTrackerCache($trackerCache);
+
+        $this->assertEquals(345, $content['PrivacyManager.ipAddressMaskLength']);
+    }
+
+    public function testSetTrackerCacheContentForSiteShouldFallbackToGlobalSettings()
+    {
+        Option::set('PrivacyManager.anonymizeReferrer', ReferrerAnonymizer::EXCLUDE_QUERY);
+
+        $trackerCache = ['existingEntry' => 'test'];
+        $this->config->setIdSite(1);
+        $content = $this->config->setTrackerCache($trackerCache);
+
+        $this->assertEquals(ReferrerAnonymizer::EXCLUDE_QUERY, $content['PrivacyManager.anonymizeReferrer']);
+    }
+
+    public function testDirectSettingGlobalConfigIsReflectedViaFallbackForSite()
+    {
+        $this->config->setIdSite(null);
```

Comment:
That call does nothing.

Is it really intented to never be able to reset the config to "all sites only" by passing `null`? Or is this call obsolete and can be removed?